### PR TITLE
Add a flexible mechanism to combine user and default plugins

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

Closes https://github.com/gazebosim/gz-sim/issues/796

## Summary
This works in conjunction with https://github.com/gazebosim/gz-sim/pull/2497 to allow specifying plugins without replacing the default list of plugins.

## Test it
Try examples in https://github.com/gazebosim/gz-sim/pull/2497/

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/gz/gui/Application.hh [hunk 1]
```diff
@@ -98,6 +98,8 @@ namespace gz::gui
       /// \sa InitializeDialogs
       public: bool LoadConfig(const std::string &_path);
 
+      public: bool LoadWindowConfig(const tinyxml2::XMLElement &_window);
```
Comment (iche033) [inline include/gz/gui/Application.hh:101]: add doxygen comments for the new functions?

File: src/Application.cc [hunk 2]
```diff
@@ -524,6 +506,37 @@ std::string Application::DefaultConfigPath()
   return this->dataPtr->defaultConfigPath;
 }
 
+std::string Application::ResolveConfigFile(const std::string &_path)
```
Comment (iche033) [inline src/Application.cc:509]: add separation line `/////////////////////////////////////////////////`

File: include/gz/gui/Application.hh (hunk 1 above)
Comment (azeey) [reply include/gz/gui/Application.hh:105]: Done in https://github.com/gazebosim/gz-gui/pull/631/commits/c2fa7925ee8c49e846957018dba9adc6b4a3202e

File: src/Application.cc (hunk 2 above)
Comment (azeey) [reply src/Application.cc:510]: Done in https://github.com/gazebosim/gz-gui/pull/631/commits/c2fa7925ee8c49e846957018dba9adc6b4a3202e

**Review by iche033 — APPROVED** [review]

**Pull request merged**